### DISCUSSION TITLE
How to integrate with plotly.py?

### DISCUSSION THREAD

**Discussion by misterkramer** [zauberzeug/nicegui#307] [Q&A] [answered]
Hi! First of all, thank you for the cool library. Question: Is it possible to use [plotly.py](https://github.com/plotly/plotly.py) with nicegui? If yes, then I would be grateful for a simple code example with interactive plot.
I can't fully understand yet - is it easy or will a feature request to nicegui be required.

**Comment by rodja** [comment, marked as answer]
Plot.ly is a great graphing library. Thanks to the extensibility of NiceGUI, it can be embedded quite easily:

```python
from nicegui import Client, ui
import plotly.express as px
import plotly.offline as pyo


@ui.page('/')
async def main(client: Client):
    df = px.data.iris()  # iris is a pandas DataFrame
    fig = px.scatter(df, x='sepal_width', y='sepal_length')

    ui.add_head_html('<script src="https://cdn.plot.ly/plotly-latest.min.js"></script>')
    ui.html('<div id="divPlotly">')
    await client.connected()

    json = fig.to_json()
    js = f'Plotly.newPlot("divPlotly", {json}); return "done";'
    await ui.run_javascript(js)
```

It would be great to simplify this even more. Feel free to add a feature request for an `ui.plotly` element. We need to think about the dependencies and implications.

**Reply by misterkramer** [reply]
Thank you very much for the quick response!
I will make a feature request for an `ui.plotly` element.
